hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java

```diff
@@ -453,4 +454,13 @@ public void closeConnection(Connection conn) throws SQLException {
 		}
 	}
 
+	private class ValidationThreadFactory implements ThreadFactory {
```

Comment:
I think you are missing some imports (at least for `ThreadFactory`) here.

Reply:
@gsmet I've added those when reviewing it and want to add a test for it as well. I'll push a PR to replicate this one when I'm done.

Reply:
FWIW, I'm not sure we'll be able to implement a test for that. If we don't, it's not a biggie IMHO.